tests/serde.rs

```diff
@@ -169,8 +180,50 @@ fn numbers() {
         to_value(&std::u64::MAX).unwrap_err();
     }
 
-    test_float!(f32);
-    test_float!(f64);
+    let big_int_serializer = Serializer::new().serialize_64_bit_numbers_as_big_int(true);
+
+    {
+        const MAX_SAFE_INTEGER: i64 = 9_007_199_254_740_991;
+
+        // u64 and i64 should serialize the same
+        test_via_into_with_config(0_i64, 0_u64, &big_int_serializer);
+        test_via_into_with_config(42_i64, 42_u64, &big_int_serializer);
+
+        // Js-numbers should also deserialize into 64 bit types
+        from_value::<i64>(JsValue::from_f64(1.0)).unwrap();
+        from_value::<i64>(JsValue::from_f64(-1.0)).unwrap();
+
+        // Test near max safe float
+        (MAX_SAFE_INTEGER + 1)
+            .serialize(&big_int_serializer)
+            .unwrap();
+        (-(MAX_SAFE_INTEGER + 1))
+            .serialize(&big_int_serializer)
+            .unwrap();
+
+        // Handle extreme values
+        std::i64::MIN.serialize(&big_int_serializer).unwrap();
+        std::i64::MAX.serialize(&big_int_serializer).unwrap();
+    }
+
+    {
+        const MAX_SAFE_INTEGER: u64 = 9_007_199_254_740_991;
+
+        test_via_into_with_config(0_u64, 0_i64, &big_int_serializer);
+        test_via_into_with_config(42_u64, 42_i64, &big_int_serializer);
+
+        from_value::<u64>(JsValue::from_f64(1.0)).unwrap();
+
+        test_via_into_with_config(
+            MAX_SAFE_INTEGER,
+            MAX_SAFE_INTEGER as i64,
+            &big_int_serializer,
+        );
+        (MAX_SAFE_INTEGER + 1)
+            .serialize(&big_int_serializer)
+            .unwrap();
+        std::u64::MAX.serialize(&big_int_serializer).unwrap();
```

Comment:
Can you also please add a few tests for conversion from a `BigInt` that doesn't actually fit into 64 bits? (you can create one from a string, it's fine to do so in tests)

It should throw an error, but I suspect right now it's not handled correctly.

Reply:
I added a js file to do validation on the BigInt's. I think it is the easiest and lowest overhead way of handling the bounds check. It replaces the `to_i64` and `to_u64` bindings used previously

Reply:
Unfortunately, the JS integration doesn't work in all wasm-bindgen modes, see [Caveats](https://rustwasm.github.io/wasm-bindgen/reference/js-snippets.html#caveats). While I agree it would be lower overhead, I'd rather keep this crate working for all output modes, as otherwise it's a pretty significant breaking change.

Reply:
As mentioned in the initial comment where I proposed the bindings idea, you can just compare the resulting `u64` with the original `BigInt` instance. It will require one more roundtrip to JS but shouldn't be _that_ bad and will be working in all modes.

Reply:
> Unfortunately, the JS integration doesn't work in all wasm-bindgen modes, see [Caveats](https://rustwasm.github.io/wasm-bindgen/reference/js-snippets.html#caveats). While I agree it would be lower overhead, I'd rather keep this crate working for all output modes, as otherwise it's a pretty significant breaking change.

I missed the lack of nodejs support while reading that page of the docs. I appreciate all of the help, I am pretty new to wasm-bindgen.